components/Buttons/Button/Button.tsx

```diff
@@ -36,20 +36,53 @@ interface GoogleAnalyticsEventPropType {
   transport?: 'beacon' | 'xhr' | 'image';
 }
 
-type ButtonProps = {
+interface ButtonProps
+  extends Omit<ButtonHTMLAttributes<HTMLButtonElement>, 'disabled'>,
+    VariantProps<typeof buttonCva> {
   /**
    * Helps track in-page `event` interactions.
    */
   analyticsObject?: GoogleAnalyticsEventPropType;
-  /**
-   * Forces the component's width as wide as its parent container's width.
-   */
-  fullWidth?: boolean;
-  /**
-   * Sets the button color theme.
-   */
-  theme?: 'primary' | 'secondary';
-} & ButtonHTMLAttributes<HTMLButtonElement>;
+}
+
+export const buttonCva = cva(
+  'inline-block font-din-condensed rounded leading-none border-4 border-solid cursor-pointer font-bold text-center uppercase py-4 px-3 whitespace-nowrap transition-all duration-200 ease-linear min-w-[175px]',
+  {
+    variants: {
+      theme: {
+        primary: 'bg-primary border-primary outline-primary text-secondary',
+        secondary: 'bg-secondary border-secondary outline-secondary text-primary',
+      },
+      fullWidth: {
+        false: null,
+        true: 'w-full',
+      },
+      disabled: {
+        false: null,
+        true: 'opacity-60 hover:cursor-not-allowed',
+      },
+    },
+    compoundVariants: [
+      {
+        theme: 'primary',
+        disabled: false,
+        class:
+          'focus-visible:bg-transparent focus-visible:text-primary hover:bg-transparent hover:text-primary',
```

Comment:
It seems that these are constantly applied, so i'd also move to L49
`focus-visible:bg-transparent hover:bg-transparent`